module/programs/helper.js

```diff
@@ -0,0 +1,724 @@
+          data: {},
+        });
+      }
+    });
+  }
+
+    /**
+   * Program details.
+   * @method
+   * @name details
+   * @param {String} programId - Program Id.
+   * @returns {Object} - Details of the program.
+   */
+
+    static details(programId) {
+        return new Promise(async (resolve, reject) => {
+          try {
+            let programData = await programsQueries.programsDocument({
+              _id: programId,
+            });
+    
+            if (!programData.length > 0) {
+              return resolve({
+                status: HTTP_STATUS_CODE.bad_request.status,
+                message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+              });
+            }
+    
+            return resolve({
+              message: CONSTANTS.apiResponses.PROGRAMS_FETCHED,
+              success: true,
+              data: programData[0],
+            });
+          } catch (error) {
+            return resolve({
+              success: false,
+              status: error.status
+                ? error.status
+                : HTTP_STATUS_CODE.internal_server_error.status,
+              message: error.message,
+            });
+          }
+        });
+      }
+
+
+     /**
+   * Add roles in program.
+   * @method
+   * @name addRolesInScope
+   * @param {String} programId - Program Id.
+   * @param {Array} roles - roles data.
+   * @returns {JSON} - Added roles data.
+   */
+
+  static addRolesInScope(programId, roles) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let programData = await programsQueries.programsDocument(
+          {
+            _id: programId,
+            scope: { $exists: true },
+            isAPrivateProgram: false,
+          },
+          ["_id"]
+        );
+
+        if (!programData.length > 0) {
+          return resolve({
+            status: HTTP_STATUS_CODE.bad_request.status,
+            message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+          });
+        }
+
+        let updateQuery = {};
+
+        if (Array.isArray(roles) && roles.length > 0) {
+          let currentRoles = await database.models.programs.find(  
+            { _id: programId },
+            { 'scope.roles': 1, _id: 0 }
+            )
+            currentRoles = currentRoles[0].scope.roles 
+
+            if(currentRoles[0] == CONSTANTS.common.ALL_ROLES){
+              updateQuery["$set"] = {
+                "scope.roles": [ CONSTANTS.common.ALL_ROLES ],
+              };
+            }
+            else{
+              let currentRolesSet = new Set(currentRoles);
+              let rolesSet = new Set(roles);
+              
+              rolesSet.forEach(role => {
+                currentRolesSet.add(role);
+              });
+              
+              currentRoles = Array.from(currentRolesSet);
+              updateQuery["$set"] = {
+                "scope.roles": currentRoles,
+              };
+              // updateQuery["$push"] = { "scope.roles": { $each: roles } }
```

Comment:
@Prajwal17Tunerlabs updateQuery["$push"] = { "scope.roles": { $each: roles } } is not working ?

Reply:
i tried, its not working so only i have commented it